cmd/earthly/main.go

```diff
@@ -619,6 +620,25 @@ func (app *earthlyApp) run(ctx context.Context, args []string) int {
 			}
 		}
 	}()
+	// Initialize log streaming early if we're passed the organization and
+	// project names as environmental variables. This will allow nearly all
+	// initialization errors to be surfaced to the log streaming service. Access
+	// to this organization and project will be verified when the stream begins.
+	if app.orgName != "" && app.projectName != "" && !app.cfg.Global.DisableLogSharing && app.logstreamUpload {
+		cloudClient, err := app.newCloudClient(cloud.WithLogstreamGRPCAddressOverride(app.logstreamAddressOverride))
+		if err != nil {
+			app.console.Warnf("Failed to initialize cloud client: %v", err)
+			return 1
+		}
+		if cloudClient.IsLoggedIn(ctx) {
+			app.console.VerbosePrintf("Logbus: setting organization %q and project %q", app.orgName, app.projectName)
+			analytics.AddEarthfileProject(app.orgName, app.projectName)
+			app.logbusSetup.SetOrgAndProject(app.orgName, app.projectName)
+			app.logbusSetup.StartLogStreamer(ctx, cloudClient)
+			logstreamURL := fmt.Sprintf("%s/builds/%s", app.getCIHost(), app.logbusSetup.InitialManifest.GetBuildId())
+			app.console.ColorPrintf(color.New(color.FgHiYellow), "Streaming logs to %s\n\n", logstreamURL)
+		}
+	}
 	app.logbus.Run().SetStart(time.Now())
```

Comment:
I don't know if we have deltas for the start time - it might be better to initialize this field before the stream begins, such that it goes in the initial manifest.

Reply:
Moved above the new block. Does that work?

Reply:
Yes, that should be good!